### PR TITLE
add check_extractable argument to high level functions

### LINKED ISSUE #350: Add check_extractable argument to high_level.extract_text
**Is your feature request related to a problem? Please describe.**
When using `pdfminer.high_level.extract_text` on some files, I get `pdfminer.pdfdocument.PDFTextExtractionNotAllowed: Text extraction is not allowed`

**Describe the solution you'd like**
Add a keyword argument `check_extractable` to `pdfminer.high_level.extract_text`, and pass it to `PDFPage.get_pages`

### PR DESCRIPTION
**Pull request by Recursing**
**Description**
Fixes #350 

**How Has This Been Tested?**
I ran extract_text on the attached pdf
[F35postaleeditabile_784_4568.pdf](https://github.com/pdfminer/pdfminer.six/files/4023402/F35postaleeditabile_784_4568.pdf)

**Checklist**

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the [README.md](../README.md) and other documentation, or I am sure that this is not necessary
- [ ] I have added a consice human-readable description of the change to [CHANGELOG.md](../CHANGELOG.md)
- [x] I have added docstrings to newly created methods and classes
- [ ] I have optimized the code at least one time after creating the initial version

### CODE REVIEW

File: pdfminer/high_level.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def extract_text_to_fp(inf, outfp, output_type='text', codec='utf-8',
                        laparams=None, maxpages=0, page_numbers=None,
                        password="", scale=1.0, rotation=0, layoutmode='normal',
                        output_dir=None, strip_control=False, debug=False,
-                       disable_caching=False, **kwargs):
+                       disable_caching=False, check_extractable=True, **kwargs):
```
Comment (igormp) [inline pdfminer/high_level.py]: CI is complaining that this line is too long

**Comment by igormp** [comment]
I'm not sure if this is the best approach to this solution, but I can't think of anything better anyway.

Would it be possible for you to fix the CI errors and add a passing test with the error mentioned in #350?

File: pdfminer/high_level.py (hunk 1 above)
Comment (Recursing) [reply pdfminer/high_level.py]: Do we need **kwargs here? They don't seem to be used

**Comment by Recursing** [comment]
What problems do you see with this approach? Should we just default check_extractable to False?
I have only two pdfs that contain actual text and trigger this issue (and two that trigger this issue but have no text), but they have a couple of images and are 650K and 880K, should I add one of them to the samples anyway?

**Comment by pietermarsman** [comment]
> What problems do you see with this approach?

See my comment [here](https://github.com/pdfminer/pdfminer.six/issues/350#issuecomment-571280593).

**Comment by pietermarsman** [comment]
I'm closing this PR and the corresponding issue (#350) because the high-level functions are for common use-cases. Changing the value of `check_extractable` is not a common use-case (imho) and the [composable AP](https://pdfminersix.readthedocs.io/en/latest/tutorials/composable.html)I should be used for this.

**Pull request closed without merging**